language/slackbot/demo_bot.js

```
@@ -0,0 +1,208 @@
+
+This is a Slack bot built using the Botkit library (http://howdy.ai/botkit). It
+runs on a Kubernetes cluster, and uses one of the Google Cloud Platform's ML
+APIs, the Natural Language API, to interact in a Slack channel. It does this in
+two respects.
+
+First, it uses the NL API to assess the "sentiment" of any message posted to
+the channel, and if the positive or negative magnitude of the statement is
+sufficiently large, it sends a 'thumbs up' or 'thumbs down' in reaction.
+
+Second, it uses the NL API to identify the 'entities' in each posted message,
+and tracks them in a database (using sqlite3).  Then, at any time you can
+query the NL slackbot to ask it for the top N entities used in the channel.
+
+The README walks through how to run the NL slackbot as an app on a
+Google Container Engine/Kubernetes cluster, but you can also just run the bot
+locally if you like.
+To do this, create a file containing your Slack token (as described in
+the README), then point 'slack_token_path' to that file when you run the script:
+
+    echo my-slack-token > slack-token
+    slack_token_path=./slack-token node demo_bot.js
+
+See the README.md in this directory for more information about setup and usage.
+
+*/
+
+'use strict';
+
+const Language = require('@google-cloud/language');
+const Storage = require('@google-cloud/storage');
+
+
+var Botkit = require('botkit');
+var fs = require('fs');
+
+var controller = Botkit.slackbot({debug: false});
+
+var sqlite3 = require('sqlite3').verbose();
+// create our database if it does not already exist.
+var db = new sqlite3.Database('slackDB.db');
+
+const util = require('util');
+
+// the number of most frequent entities to retrieve from the db on request.
+const NumEnts = 20;
+// The magnitude of sentiment of a posted text above which the bot will respond.
+const SentimentThresh = 30;
+
+if (!process.env.slack_token_path) {
+  console.log('Error: Specify slack_token_path in environment');
+  process.exit(1);
```

NIT: Throw an error instead of logging and exiting, e.g.

```js
throw new Error('Specify slack_token_path in environment!');
```